src/operation/insert.rs

```diff
@@ -54,46 +60,26 @@ impl<'a> OperationWithDefaults for Insert<'a> {
     const NAME: &'static str = "insert";
 
     fn build(&mut self, description: &StreamDescription) -> Result<Command<Self::Command>> {
+        let max_doc_size: usize = Checked::new(description.max_bson_object_size).try_into()?;
+        let max_message_size: usize =
+            Checked::new(description.max_message_size_bytes).try_into()?;
+        let max_operations: usize = Checked::new(description.max_write_batch_size).try_into()?;
+
+        let mut command_body = rawdoc! { Self::NAME: self.ns.coll.clone() };
+        let options = bson::to_raw_document_buf(&self.options)?;
+        extend_raw_document_buf(&mut command_body, options)?;
+
+        let max_document_sequence_size =
+            max_message_size - OP_MSG_OVERHEAD_BYTES - command_body.as_bytes().len();
```

Comment:
Suggest using `Checked` to prevent possible underflow.

```suggestion
        let max_document_sequences_size : usize =
            (Checked::new(max_message_size) - OP_MSG_OVERHEAD_BYTES - command_body.as_bytes().len()).try_into()?;
```